tests/phpunit/tests/blocks/editor.php

```diff
@@ -720,4 +720,36 @@ public function data_block_editor_rest_api_preload_adds_missing_leading_slash()
 			),
 		);
 	}
+
+	/**
+	 * @ticket 61641
+	 */
+	public function test_get_block_editor_settings_block_bindings_sources() {
+		$block_editor_context = new WP_Block_Editor_Context();
+		register_block_bindings_source(
+			'test/source-one',
+			array(
+				'label'              => 'Source One',
+				'get_value_callback' => function () {},
+				'uses_context'       => array( 'postId' ),
+			)
+		);
+		register_block_bindings_source(
+			'test/source-two',
+			array(
+				'label'              => 'Source Two',
+				'get_value_callback' => function () {},
+			)
+		);
+		$settings        = get_block_editor_settings( array(), $block_editor_context );
+		$exposed_sources = $settings['blockBindingsSources'];
```

Comment:
Should we deregister custom sources here so they don't leak to other tests?

Reply:
You're right we should do that. I've done it in [this commit](https://github.com/WordPress/wordpress-develop/pull/7020/commits/a18de617e53c0843cb3874cc4539b52296a9659b), plus unregistering them from other tests which were causing some issues.

Reply:
This looks good, thank you. I left comments in other places as it's more nuanced sometimes. Anyway, it's a minor thing that it's easy to address.